pipeline/extract.py

```diff
@@ -0,0 +1,198 @@
+'''File used for extracting information from bandcamp's website and API'''
+
+from time import sleep
+import json
+import logging
+import asyncio
+import aiohttp
+import requests as req
+from bs4 import BeautifulSoup
+
+
+BANDCAMP_SALES_URL = "https://bandcamp.com/api/salesfeed/1/get_initial"
+MAX_TIMEOUT_SECONDS = 20
+
+
+def get_sale_data_from_api(site_url: str = BANDCAMP_SALES_URL,
+                           max_timeout: int = MAX_TIMEOUT_SECONDS) -> dict:
+    '''Get content from the specified api endpoint'''
+    sleep(10)
+    try:
+        response = req.get(site_url, timeout=max_timeout)
+
+        if response.status_code == 200:
+            response_contents = response.json()
+
+            return response_contents
+        logging.error("Failed to retrieve sale data from API. HTTP Status code: %s",
+                      response.status_code)
+
+    except req.exceptions.RequestException as e:
+        logging.error("A request error has occurred: %s", e)
+    except TimeoutError as e:
+        logging.error("A timeout error has occurred: %s", e)
+    return None
+
+
+def save_to_json(list_to_be_written: list, filename: str) -> None:
```

Comment:
```suggestion
def save_to_json(sales_data: list[dict], filename: str) -> None:
```
Changing the typehint and parameter to something similar to this would be more accurate, I think.